src/com/palmergames/bukkit/towny/utils/PermissionUtil.java

```diff
@@ -0,0 +1,155 @@
+package com.palmergames.bukkit.towny.utils;
+
+import com.palmergames.bukkit.towny.TownyAPI;
+import com.palmergames.bukkit.towny.TownyFormatter;
+import com.palmergames.bukkit.towny.command.PlotCommand;
+import com.palmergames.bukkit.towny.exceptions.TownyException;
+import com.palmergames.bukkit.towny.object.PermissionData;
+import com.palmergames.bukkit.towny.object.gui.EditGUI;
+import com.palmergames.bukkit.towny.object.gui.PermissionGUI;
+import com.palmergames.bukkit.towny.object.Resident;
+import com.palmergames.bukkit.towny.object.TownBlock;
+import com.palmergames.bukkit.towny.object.TownyPermission.ActionType;
+import com.palmergames.bukkit.towny.object.Translation;
+import com.palmergames.bukkit.util.Colors;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.Material;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.inventory.meta.SkullMeta;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map.Entry;
+
+public class PermissionUtil {
```

Comment:
I think this might need to be renamed to something like GUIPermissionUtil or PlotPermissionGUIUtil, so that its not confused with "permission node"-related things.